lib/engine/abilities.rb

```diff
@@ -10,46 +10,53 @@
 module Engine
   module Abilities
     def init_abilities(abilities)
-      @abilities = {}
+      @abilities = []
 
       (abilities || []).each do |ability|
         klass = Ability::Base.type(ability[:type])
         ability = Object.const_get("Engine::Ability::#{klass}").new(**ability)
-        raise 'Duplicate abilities detected' if @abilities[ability.type]
-
         ability.owner = self
-        @abilities[ability.type] = ability
+        @abilities << ability
       end
     end
 
-    def abilities(type, time = nil)
-      return nil unless (ability = @abilities[type])
+    def ability_by_type(type)
```

Comment:
i don't like how we can't handle duplicate abilities i wonder if there's a way around  this?

Reply:
How about this? f2844fdd

I think code that calls `abilities` will need to be changed when we have some corp/company that has duplicate abilities, but `abilities` itself should do the right thing.

Reply:
These two lines are included for backward compatibility without having to change other code, ideally we'd just return `active_abilities`; that could be fixed when we need to actually implement duplicate abilities - https://github.com/tobymao/18xx/pull/1396/files#diff-9abc48bc27003a6922f597d541dc4f3eR48-R49